cloudinit/config/cc_growpart.py

```diff
@@ -92,6 +93,52 @@ class RESIZE(object):
 
 LOG = logging.getLogger(__name__)
 
+@lru_cache()
+def is_lvm_lv(devpath):
+    if util.is_Linux():
+        # all lvm lvs will have a realpath as a 'dm-*' name.
+        rpath = os.path.realpath(devpath)
+        if not os.path.basename(rpath).startswith("dm-"):
+            return False
+        out, _ = subp.subp("udevadm", "info", devpath)
+        # lvs should have DM_LV_NAME=<lvmuuid> and also DM_VG_NAME
+        return 'DM_LV_NAME=' in out
+    else:
+        False
+
+@lru_cache()
+def get_pvs_for_lv(devpath):
+    myenv = os.environ.copy()
+    myenv = {'LANG': 'C'}
+
+    if util.is_Linux():
+        try:
+            (out, _err) = subp.subp(["lvm", "lvs", devpath, "-options=vgname",
+                                     "--noheadings"], update_env=myenv)
+            vgname = out.strip()
+        except subp.ProcessExecutionError as e:
+            if e.exit_code != 0:
+                util.logexc(LOG, "Failed: can't get Volume Group information "
+                            "from %s", devpath)
+                raise ResizeFailedException(e) from e
+
+        try:
+            (out, _err) = subp.subp(["lvm", "vgs", vgname, "-options=pvname",
+                                     "--noheadings"], update_env=myenv)
+            pv = out.strip().splitlines()
+            if len(pvs) > 1:
+                util.logexc(LOG, "Failed: cloud-init currently supports only "
+                            "one Physical Volume per Volume Group")
```

Comment:
as i said in IRC, i think that this should be an error if the user explicitly asked for resizing, but if this is just defaults, then debug or info "do not know how to resize multiple pvs".